### PR TITLE
SRS checker plugin 

### PR DESCRIPTION
**Pull request by lbesnard**
- add SRS SST tests and unittests

### CODE REVIEW

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py [hunk 1]
```diff
@@ -936,7 +935,147 @@ def test_check_acknowledgement(self):
...
+        self.addCleanup(nc_dataset.close)
+        return nc_dataset
+
+    def setUp(self):
+        '''
+        Initialize the dataset
+        '''
+        self.srs          = IMOSSRSBaseCheck()
+        self.good_dataset = self.load_dataset(STATIC_FILES['good_data_srs'])
+        self.bad_dataset  = self.load_dataset(STATIC_FILES['bad_data_srs'])
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py]: Again, it would be a bit more obvious if these were just `self.srs_good_data` and `self.srs_bad_data` 
(TBD for the imos tests too...)

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py [hunk 2]
```diff
@@ -936,7 +935,147 @@ def test_check_acknowledgement(self):
...
+
+        for result in ret_val:
+            self.assertTrue(result.value)
+
+        ret_val = self.srs.check_variable_attribute_type(self.bad_dataset)
+
+        for result in ret_val:
+            if result.value == False: # check l2p_flags:valid_min
+                self.assertFalse(result.value)
+    def test_check_data_variable_present(self):
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py]: Add blank line above.

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py [hunk 3]
```diff
@@ -936,7 +935,147 @@ def test_check_acknowledgement(self):
...
+        self.assertEqual(len(ret_val), 15)
+        self.assertEqual(len(failed_var), 1)
+        self.assertEqual(set(failed_var), set(['random_var']))
+
+    @unittest.skipUnless(netCDF4.__netcdf4libversion__ >= '4.3',
+                         'requires netCDF4 library version >= 4.3')
+    def test_check_fill_value(self):
+        ret_val = self.srs.check_fill_value(self.bad_dataset)
+        failed_var = [r.name[1] for r in ret_val if not r.value]
+        self.assertEqual(failed_var, ['random_var'])
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py]: Your code is failing this test. I think it should be the 'lon' variable, instead of 'random_var'.

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py [hunk 4]
```diff
@@ -1,14 +1,13 @@
 #!/usr/bin/env python
 
 from cc_plugin_imos.imos import IMOS1_3Check, IMOS1_4Check
+from cc_plugin_imos.srs import  IMOSSRSBaseCheck
```
Comment (lbesnard) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py]: I sure can. But to mention that in that case, it also would make sense to put the class TestUtils in a different file

File: lib/cc_plugin_imos/cc_plugin_imos/util.py [hunk 5]
```diff
@@ -189,6 +190,7 @@ def find_data_variables(dataset, coordinate_variables, ancillary_variables):
         if var not in coordinate_variables and var not in \
             ancillary_variables and var.dimensions and var not in \
             auxiliary_coordinate_variables \
+            and not hasattr(var, 'flag_meanings') \
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/util.py]: Yes, that looks ok. It would be nice to separate this change (which also affects the IMOS checkers) from your addition of the SRS checker. Could you put it in a separate commit? (if not a separate PR)

Comment (lbesnard) [inline lib/cc_plugin_imos/cc_plugin_imos/util.py]: forget what I sed.

**Comment by lbesnard** [comment]
@mhidas  FYI https://help.github.com/articles/why-are-my-commits-in-the-wrong-order/ 
The commits appeared in wrong order in this PR, however, they should be right if you pull the branch

```
commit 8c034db0715ae05b185b2bb0fe4a49c9a1653c2c
Author: lbesnard <laurent.besnard@utas.edu.au>
Date:   Wed Nov 9 12:45:58 2016 +1100

    IMOS CHECKER - unittest on temp netcdf files
    
     * unittests use to run on NetCDF files generated from CDL. The NetCDF
       were written in the checker plugin folder, which created permission
       issues for ncgen to create those files since we moved the checker to
       an egg version
     * the netcdf files are now created and deleted in the temp directory.

commit 1117364dd9cb391be5b8a180611bbb9fc085bfca
Author: lbesnard <laurent.besnard@utas.edu.au>
Date:   Mon Oct 24 13:26:20 2016 +1100

    SRS checker plugin
    
     * add SRS SST tests and unittests

commit feaecb4b8fab798e5e5afcea07e48e006aac733d
Author: lbesnard <laurent.besnard@utas.edu.au>
Date:   Wed Nov 9 12:33:09 2016 +1100

    NETCDF CHECKER - remove non data var from data_variable function output
    
     * modify find_data_variables function in util.py to remove from a list
       of data variables variables with flag_meanings attribute
```

Happy to chat regarding the last one

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py [hunk 6]
```diff
@@ -56,12 +58,21 @@ def load_dataset(self, nc_dataset):
         self.addCleanup(nc_dataset.close)
         return nc_dataset
 
+    @classmethod
+    def setUpClass(cls):
+        cls.static_files = static_files_testing()
+
+    @classmethod
+    def tearDownClass(cls):
+        for file_path in [cls.static_files[v] for v in cls.static_files]:
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_imos.py]:
```python
for file_path in cls.static_files.values():
```
Looks a bit clearer to me.

File: lib/cc_plugin_imos/cc_plugin_imos/tests/test_srs.py [hunk 7]
```diff
@@ -42,13 +44,22 @@ def load_dataset(self, nc_dataset):
         self.addCleanup(nc_dataset.close)
         return nc_dataset
 
+    @classmethod
+    def setUpClass(cls):
+        cls.static_files = static_files_testing()
+
+    @classmethod
+    def tearDownClass(cls):
+        for file_path in [cls.static_files[v] for v in cls.static_files]:
```
Comment (mhidas) [inline lib/cc_plugin_imos/cc_plugin_imos/tests/test_srs.py]: same here
